functests/2_performance_update/updating_profile.go

```diff
@@ -55,6 +56,67 @@ var _ = Describe("[rfe_id:28761][performance] Updating parameters in performance
 		Expect(err).ToNot(HaveOccurred())
 	})
 
+	Context("Verify GloballyDisableIrqLoadBalancing Spec field", func() {
+		It("[test_id:36150] Verify that IRQ load balancing is enabled/disabled correctly", func() {
+			irqLoadBalancingDisabled := profile.Spec.GloballyDisableIrqLoadBalancing != nil && *profile.Spec.GloballyDisableIrqLoadBalancing
+
+			profileIsolatedCPUs := string(*profile.Spec.CPU.Isolated)
+			isolatedCPUsMask, err := components.CPUListToHexMask(profileIsolatedCPUs)
+			Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("failed to list in Hex %s: %v", profileIsolatedCPUs, err))
+			isolatedCPUs, err := strconv.ParseUint(isolatedCPUsMask, 16, 64)
+			Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("failed to parse %s: %v", isolatedCPUsMask, err))
+
+			var expectedBannedCPUs uint64
+			if irqLoadBalancingDisabled {
+				expectedBannedCPUs = isolatedCPUs
+			}
+
+			for _, node := range workerRTNodes {
+				bannedCPUs, err := nodes.BannedCPUs(node)
+				Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("failed to extract the banned CPUs from node %s: %v", node.Name, err))
+				Expect(bannedCPUs).To(Equal(expectedBannedCPUs))
+			}
+
+			By("Modifying profile")
+			initialProfile = profile.DeepCopy()
+
+			irqLoadBalancingDisabled = !irqLoadBalancingDisabled
+			if irqLoadBalancingDisabled {
+				expectedBannedCPUs = isolatedCPUs
+			} else {
+				expectedBannedCPUs = 0
+			}
+			profile.Spec.GloballyDisableIrqLoadBalancing = &irqLoadBalancingDisabled
+
+			By("Updating the performance profile")
+			Expect(testclient.Client.Update(context.TODO(), profile)).ToNot(HaveOccurred())
+			defer func() { // return initial configuration
+				spec, err := json.Marshal(initialProfile.Spec)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(testclient.Client.Patch(context.TODO(), profile,
+					client.RawPatch(
+						types.JSONPatchType,
+						[]byte(fmt.Sprintf(`[{ "op": "replace", "path": "/spec", "value": %s }]`, spec)),
+					),
+				)).ToNot(HaveOccurred())
+			}()
+
+			Eventually(func() error {
```

Comment:
with the added waits(), we can use the same loop as line 74 above, which is nice for consistency

Reply:
What do you mean? 
The first loop doesn't have to wait for anything, this one does.
If you have a nicer way to code it, please point me to it, I will gladly improve (I am not mastering the Ginko stuff yet)

Reply:
ok, we can keep as it is.

Reply:
Please elaborate more, I want to at lest know it. Can you add more context to the "Wait()" ?
Also we should have conventions.

Reply:
Setting up conventions (but most notably enforce them) would be interesting.
Thing is the usage of Eventually() here suggests that changes to the cluster made in the previous steps of the tests (e.g. change the profile) may take some time to be accepted, and then the state of the system will Eventually() be what's expected - otherwise the test will fail.
That works.
Another way is to wait explicitely for the changes to be accepted by the cluster, hence after each stage is performed the state of the cluster is known (or otherwise a previous expectation failed, letting the test fail). Once the state of the cluster is known, there is often no need to use Eventually(), which may hide bugs.
But again is not a clear cut or an obvious choice: it depends on how the test is written, and sometimes depends also on other factos (e.g. are we trying to cut down the test execution time?)

Reply:
Thanks for the explanation, so basically if we change the profile we should test that the PAO updated the profile and NTO also updated the tuned config AND finished to update the node. I think it will be doable if we'll have all this info in the PAO status section. Then we can simply wait for the status to be updated.  This is what you have in mind?

Reply:
pretty much, yes. We can wait directly on MCP/NTO statuses if required or simpler. Anyway, no big deal for this PR.